manifests/init.pp

```diff
@@ -47,26 +86,39 @@
 #   will move files prior to installation that are known to cause file locks.
 #
 class puppet_agent (
-  $arch                  = $::architecture,
-  $collection            = $::puppet_agent::params::collection,
-  $is_pe                 = $::puppet_agent::params::_is_pe,
-  $manage_pki_dir        = true,
-  $manage_repo           = true,
-  $package_name          = 'puppet-agent',
-  $package_version       = undef,
-  $service_names         = $::puppet_agent::params::service_names,
-  $source                = undef,
-  $install_dir           = undef,
-  $disable_proxy         = false,
-  $install_options       = [],
-  $skip_if_unavailable   = 'absent',
-  $msi_move_locked_files = false,
+  $arch                    = $::architecture,
+  $collection              = $::puppet_agent::params::collection,
+  $is_pe                   = $::puppet_agent::params::_is_pe,
+  $manage_pki_dir          = true,
+  $manage_repo             = true,
+  $package_name            = 'puppet-agent',
+  $package_version         = undef,
+  $service_names           = $::puppet_agent::params::service_names,
+  $source                  = undef,
+  $absolute_source         = undef,
+  $yum_source              = 'http://yum.puppet.com',
```

Comment:
insecure make me sad.

Reply:
Agreed, we found out that EL-5 machines will fail if the source is set to https. I'm also sad about insecure _but_ we do sign our packages. Hopefully even if someone did manage an attack the upgrade would still not do anything since any packages provided by an attacker wouldn't be signed.